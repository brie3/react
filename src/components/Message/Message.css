.message {
    background-color: lightblue;
    border-radius: 10px;
    padding: 10px;
    margin: 10px 0;
    width: max-content;
}

.message-person {
    align-self: flex-end;
}
