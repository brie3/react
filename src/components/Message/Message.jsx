import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import { UserContext } from "../UserContext";
import("./Message.sass");

export const messageType = {
    name: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired
};
export class Message extends Component {
    static propTypes = messageType;
    static contextType = UserContext;
    render() {
        const { name, content } = this.props;
        const msgClass = classNames("message", {
            "message-person": name !== "Robot"
        });
        return (
            <div className={msgClass}>
                <b>{name}</b>: {content}
                <b>{this.context.user.name}</b>
            </div>
        );
    }
}
