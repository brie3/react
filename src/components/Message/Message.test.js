import React from "react";
import renderer from "react-test-renderer";
import { Message } from "./Message";
import { shallow } from "enzyme";

describe("<Message>", () => {
    it(" renders our content", () => {
        const props = { name: "My user", content: "My content" };

        const element = renderer.create(<Message {...props} />).toJSON();
        expect(element).toMatchSnapshot();
    });
    it("message without name", () => {
        const props = { content: "My content" };
        const element = shallow(<Message {...props} />);
        expect(element.contains(<b>"Anonymous"</b>)).toBe(true);
    });
    it("message with name", () => {
        const props = { name: "My user" };
        const element = shallow(<Message {...props} />);
        expect(element.contains(<b>{props.name}</b>)).toBe(true);
    });
});
