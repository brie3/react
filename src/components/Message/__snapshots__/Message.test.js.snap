// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message>  renders our content 1`] = `
<div
  className="MuiTypography-root MuiPaper-root MuiPaper-elevation6 MuiSnackbarContent-root message message-person MuiTypography-body2"
  role="alert"
>
  <div
    className="MuiSnackbarContent-message"
  >
    <div>
      <p>
        My content
      </p>
      <small>
        created on 
         by 
        <b>
          My user
        </b>
      </small>
    </div>
  </div>
  <div
    className="MuiSnackbarContent-action"
  >
    <svg
      aria-hidden="true"
      aria-label="Delete Message"
      className="MuiSvgIcon-root message-delete"
      focusable="false"
      onClick={[Function]}
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M14.12 10.47L12 12.59l-2.13-2.12-1.41 1.41L10.59 14l-2.12 2.12 1.41 1.41L12 15.41l2.12 2.12 1.41-1.41L13.41 14l2.12-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4zM6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9z"
      />
    </svg>
  </div>
</div>
`;
